examples/frequentist_notebook_jax.py

```diff
@@ -90,156 +92,185 @@ def match_date(start_date):
+    qm.get_relative_growths(confints_estimates[1], n_variants=n_variants_effective),
+):
+    print(f"  {variant}: {float(m):.2f} ({float(l):.2f} – {float(u):.2f})")
+# -
+
+
+# We can propagate this uncertainty to the observed values. Let's generate confidence bands around the fitted lines and predict the future behaviour.
+
+# +
+ys_fitted_confint = qm.get_confidence_bands_logit(
+    theta_star,
+    n_variants=n_variants_effective,
+    ts=ts_lst_scaled,
+    covariance=covariance_scaled,
 )
 
+
 ## compute predicted values and confidence bands
 horizon = 60
 ts_pred_lst = [jnp.arange(horizon + 1) + tt.max() for tt in ts_lst]
-ts_pred_lst_scaled = [(x - t_min) / (t_max - t_min) for x in ts_pred_lst]
+ts_pred_lst_scaled = time_scaler.transform(ts_pred_lst)
 
-y_pred_lst = qm.fitted_values(
-    ts_pred_lst_scaled, theta=solution.x, cities=cities, n_variants=len(variants2)
+ys_pred = qm.fitted_values(
+    ts_pred_lst_scaled, theta=theta_star, cities=cities, n_variants=n_variants_effective
 )
-# ... and because of https://github.com/cbg-ethz/covvfit/issues/24
-# we need to transpose again
-y_pred_lst = [y.T[1:] for y in y_pred_lst]
-
-y_pred_lst_confint = qm.get_confidence_bands_logit(
-    solution.x, len(variants2), ts_pred_lst_scaled, covariance_scaled
+ys_pred_confint = qm.get_confidence_bands_logit(
+    theta_star,
+    n_variants=n_variants_effective,
+    ts=ts_pred_lst_scaled,
+    covariance=covariance_scaled,
 )
+# -
 
+# ## Plot
+#
+# Finally, we plot the abundance data and the model predictions. Note that the 0th element in each array corresponds to the artificial "other" variant and we decided to plot only the explicitly defined variants.
 
-# -
+# +
+colors = [plot_ts.COLORS_COVSPECTRUM[var] for var in variants_investigated]
 
-# ## Plotting functions
 
-plot_fit = plot_ts.plot_fit
-plot_complement = plot_ts.plot_complement
-plot_data = plot_ts.plot_data
-plot_confidence_bands = plot_ts.plot_confidence_bands
+figure_spec = plot.arrange_into_grid(len(cities), axsize=(4, 1.5), dpi=350, wspace=1)
 
-# ## Plot
 
-# +
-colors_covsp = plot_ts.colors_covsp
-colors = [colors_covsp[var] for var in variants]
-fig, axes_tot = plt.subplots(4, 2, figsize=(15, 10))
-axes_flat = axes_tot.flatten()
-
-for i, city in enumerate(cities):
-    ax = axes_flat[i]
-    # plot fitted and predicted values
-    plot_fit(ax, ts_lst[i], y_fit_lst[i], variants, colors)
-    plot_fit(ax, ts_pred_lst[i], y_pred_lst[i], variants, colors, linetype="--")
-
-    #     # plot 1-fitted and predicted values
-    plot_complement(ax, ts_lst[i], y_fit_lst[i], variants)
-    #     plot_complement(ax, ts_pred_lst[i], y_pred_lst[i], variants, linetype="--")
-    # plot raw deconvolved values
-    plot_data(ax, ts_lst[i], ys_lst2[i], variants, colors)
-    # make confidence bands and plot them
-    conf_bands = y_fit_lst_confint[i]
-    plot_confidence_bands(
+def plot_city(ax, i: int) -> None:
+    def remove_0th(arr):
```

Comment:
Note that we don't define any auxiliary arrays for the "variants under investigation"/"real variants" anymore: we always work with the "effective variants", i.e., with an additional auxiliary "other" variant.

Only at the end, when  we want to plot only the "real variants", we remove it using this slicing function.
Doing it in this way, we have fewer arrays to remember about.